### PR TITLE
Add documentation for auxiliary/scanner/http/http_traversal module

### PR DESCRIPTION
**Pull request by errorxyz**
This commit adds a documentation for auxiliary/scanner/http/http_traversal module

Related to https://github.com/rapid7/metasploit-framework/issues/12389

### CODE REVIEW

File: documentation/modules/auxiliary/scanner/http/http_traversal.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+1. Do: `run`
+
+## Options
+
+1. DATA - HTTP body data
+1. DEPTH - Traversal depth (default: 5)
+1. FILELIST - file containing list of files to bruteforce for (default: /usr/share/metasploit-framework/data/wordlists/sensitive_files.txt)
+1. METHOD - HTTP request method to use (default: GET)
+1. PATH - Vulnerable path. Ex: /foo/index.php?pg= (default: /)
+1. PATTERN - Regexp pattern to determine directory traversal (default: ^HTTP/\d\.\d 200)
```
Comment (jheysel-r7) [inline documentation/modules/auxiliary/scanner/http/http_traversal.md:48]: Please change the options to follow the format outlined here:
https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/module_doc_template.md

Comment (errorxyz) [reply documentation/modules/auxiliary/scanner/http/http_traversal.md]: Fixed the format

**Comment by bwatters-r7** [comment]
```
msf6 > use auxiliary/scanner/http/http_traversal 
msf6 auxiliary(scanner/http/http_traversal) > set rhost 10.5.134.129
rhost => 10.5.134.129
msf6 auxiliary(scanner/http/http_traversal) > set path /?p=
path => /?p=
msf6 auxiliary(scanner/http/http_traversal) > run

[*] Running action: CHECK...
[+] Found trigger: ../
[+] Directory traversal found: ../
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/http/http_traversal) > 

```

**Comment by bwatters-r7** [comment]
# Release Notes:
This PR adds documentation for the auxiliary/scanner/http/http_traversal module.

**Pull request merged**